test/instrumentation/modules/@aws-sdk/fixtures/use-s3-v3.js

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright Elasticsearch B.V. and other contributors where applicable.
+ * Licensed under the BSD 2-Clause License; you may not use this file except in
+ * compliance with the BSD 2-Clause License.
+ */
+
+'use strict'
+
+// Run a single scenario of using the S3 client (callback style) with APM
+// enabled. This is used to test that the expected APM events are generated.
+// It writes log.info (in ecs-logging format, see
+// https://github.com/trentm/go-ecslog#install) for each S3 client API call.
+//
+// This script can also be used for manual testing of APM instrumentation of S3
+// against a real S3 account. This can be useful because tests are done against
+// https://github.com/localstack/localstack that *simulates* S3 with imperfect
+// fidelity.
+//
+// Auth note: By default this uses the AWS profile/configuration from the
+// environment. If you do not have that configured (i.e. do not have
+// "~/.aws/...") files, then you can still use localstack via setting:
+//    unset AWS_PROFILE
+//    export AWS_ACCESS_KEY_ID=fake
+//    export AWS_SECRET_ACCESS_KEY=fake
+// See also: https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html
+//
+// Usage:
+//    # Run against the default configured AWS profile, creating a new bucket
+//    # and deleting it afterwards.
+//    node use-s3-v3.js | ecslog
+//
+//    # Testing against localstack.
+//    docker run --rm -it -e SERVICES=s3 -p 4566:4566 localstack/localstack
+//    TEST_ENDPOINT=http://localhost:4566 node use-s3-v3.js | ecslog
+//
+//    # Use TEST_BUCKET_NAME to re-use an existing bucket (and not delete it).
+//    # For safety the bucket name must start with "elasticapmtest-bucket-".
+//    TEST_BUCKET_NAME=elasticapmtest-bucket-1 node use-s3.js | ecslog
+//
+// Output from a sample run is here:
+// https://gist.github.com/trentm/c402bcab8c0571f26d879ec0bcf5759c
+
+const apm = require('../../../../..').start({
+  serviceName: 'use-s3-v3',
+  captureExceptions: false,
+  centralConfig: false,
+  metricsInterval: 0,
+  cloudProvider: 'none',
+  stackTraceLimit: 4, // get it smaller for reviewing output
+  logLevel: 'info'
+})
+
+const crypto = require('crypto')
+const assert = require('assert')
+const {
+  S3Client,
+  ListBucketsCommand,
+  CreateBucketCommand,
+  DeleteBucketCommand,
+  PutObjectCommand,
+  GetObjectCommand,
+  DeleteObjectCommand,
+  waitUntilBucketExists,
+  waitUntilObjectExists
+} = require('@aws-sdk/client-s3')
+
+const TEST_BUCKET_NAME_PREFIX = 'elasticapmtest-bucket-'
+
+// ---- support functions
+
+// https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-s3/index.html
+async function useS3v3 (s3Client, bucketName) {
+  const region = s3Client.config.region
```

Comment:
I've confirmed that although the `region` is passed as string in the constructor call the config property is transformed to an async function. This does not happen with the `useArnRegion` configuration property.

I've applied your change but since [the documentation](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-s3/interfaces/s3clientconfig.html) says we may get a plain string for `region` I think is best to keep the type checking in the instrumentation.